### DISCUSSION TITLE
UTM - macOS set computer serial number

### DISCUSSION THREAD

**Discussion by bpourcel** [utmapp/UTM#4488] [UTM Help (macOS)]
Hi,

We use Apple business Manager and we have DEP active.
I'd like to use UTM to test DEP process and MDM enrollment.

How can i change the serial number of the macOS UTM VM for a SN present in our ABM ?

Thanks
Bertrand

**Comment by kvellano** [comment]
Bump! Would love to see this feature available if possible. User-initiated Enrollment testing suffices but it would be awesome to have device enrollment capabilities.

**Comment by macownersclub** [comment]
Bump!

**Comment by NajiNasimi-AIYou** [comment]
Something similar to Parallels Desktop Pro would be preferable. From my understanding since UTM does not modify any flags and only runs natively, we would not be able to get flags as such. Only way around would be to go with docker implementation of this. 

Enter the following two system flags:
```
devices.smbios.serial=<serial>

devices.mac_hw_model=<mac_model>
```

For example:

```
devices.smbios.serial="C02XK123456"

devices.mac_hw_model="Macmini7,1"
```

**Reply by jeff4transformance** [reply]
I see HardwareModel and Serial fields in the config.plist file for my VM. Is it possible to use that to add the serial and model ID? Anyone know the syntax for that if it is possible?

**Reply by jeff4transformance** [reply]
Maybe the Serial field is for a serial connection and not specifying a serial number.

**Reply by someguyinbaltimore** [reply]
I edited it for kicks and it broke the VM.

**Comment by woodsy900** [comment]
I am guessing this has not been implemented? I would love this too for this exact reason! I am test driving a Mac for a while and having this option to be able to do Autopilot Kiosk deployment tests would be awesome!

**Comment by woodsy900** [comment]
So I was digging around in the discord and found that someone had added in the vm settings under QEMU Arguments the new arguement of

`-smbios
type=1,serial=xxxx`

They provided a screenshot of the SerialNumber being displayed in Cmd

**Comment by mjlodge** [comment]
But on Apple Silicon QEMU is not used... so still need this capability when using apple virtualization.

**Comment by truatpasteurdotfr** [comment]
https://discussions.apple.com/thread/255669818?sortBy=rank 
https://kb.parallels.com/123455

Until the Apple framework allows it, I don't see it happening anytime soon.
